src/Transport/AzureServiceBusTransportTransaction.cs

```diff
@@ -0,0 +1,95 @@
+    /// between the incoming message (if available) and the outgoing messages dispatched.
+    /// </summary>
+    public sealed class AzureServiceBusTransportTransaction : IDisposable
+    {
+        /// <summary>
+        /// Creates a new instance of an <see cref="AzureServiceBusTransportTransaction"/> with an optional transport transaction.
+        /// </summary>
+        /// <remarks>The current instance registers itself automatically in the transport transaction.</remarks>
+        /// <param name="transportTransaction">An optional transport transaction instance.</param>
+        public AzureServiceBusTransportTransaction(TransportTransaction? transportTransaction = null)
+        {
+            TransportTransaction = transportTransaction ?? new TransportTransaction();
+            TransportTransaction.Set(this);
+        }
+
+        /// <summary>
+        /// Creates a new instance of an <see cref="AzureServiceBusTransportTransaction"/> with the provided connection information.
+        /// The connection information is necessary in cases cross entity transactions are in use.
+        /// </summary>
+        /// <remarks>The current instance registers itself automatically in the transport transaction.</remarks>
+        /// <param name="serviceBusClient">The service bus client to be used for creating senders.</param>
+        /// <param name="incomingQueuePartitionKey">The incoming queue partition key to be used to set <see cref="ServiceBusMessage.TransactionPartitionKey"/></param>
+        /// <param name="transactionOptions">The transaction options to be used when the underlying committable transaction is created.</param>
+        /// <param name="transportTransaction">An optional transport transaction instance.</param>
+        public AzureServiceBusTransportTransaction(ServiceBusClient serviceBusClient, string incomingQueuePartitionKey,
+            TransactionOptions transactionOptions, TransportTransaction? transportTransaction = null)
+        : this(transportTransaction)
+        {
+            ServiceBusClient = serviceBusClient;
+            IncomingQueuePartitionKey = incomingQueuePartitionKey;
+            this.transactionOptions = transactionOptions;
+        }
+
+        /// <summary>
+        /// Gets the currently owned transport transaction.
+        /// </summary>
+        public TransportTransaction TransportTransaction { get; }
+
+        /// <summary>
+        /// Gets the <see cref="CommittableTransaction"/> in case cross entity transactions are used.
+        /// </summary>
+        /// <returns>A committable transaction or null.</returns>
+        /// <remarks>The committable transaction is lazy initialized as late as possible to make sure
+        /// the transaction timeout is only started when the transaction is really needed.</remarks>
+        internal CommittableTransaction? CommittableTransaction
```

Comment:
I wonder if there is a better name for this property that is more descriptive of when the transaction is used, rather than just copying the type name (e.g. CrossEntityTransaction?)

Reply:
Aren't those two different though? 

`CrossEntityTransaction` describes a concept of messages distributed between entities under a transaction. `CommittableTransaction` describes a transaction execution mode. If anything, this property would be called `Transaction`.

Reply:
Switched to transaction. I also decided to only expose the base class and no longer the committable version